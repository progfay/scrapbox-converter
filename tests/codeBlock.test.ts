/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../src'
import pretty from 'pretty'

describe('codeBlock', () => {
  it('Simple codeBlock', () => {
    const input = `Code Block
code:index.js
 function () {
   alert(document.location.href)
   console.log("hello")
   // You can also write comments!
 }`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('CodeBlock with not registered language', () => {
    const input = `Code Block
code:hoge
 this.language.is()`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
