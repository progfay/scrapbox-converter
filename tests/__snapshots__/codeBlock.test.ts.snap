// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codeBlock CodeBlock with not registered language 1`] = `
"<article>
  <h1 class=\\"title\\">Code&nbsp;Block</h1>
  <section><pre><code><span class=\\"hljs-keyword\\">this</span>.language.<span class=\\"hljs-keyword\\">is</span>()</code></pre>
  </section>
</article>"
`;

exports[`codeBlock Simple codeBlock 1`] = `
"<article>
  <h1 class=\\"title\\">Code&nbsp;Block</h1>
  <section><pre><code><span class=\\"hljs-function\\"><span class=\\"hljs-keyword\\">function</span> (<span class=\\"hljs-params\\"></span>) </span>{
  alert(<span class=\\"hljs-built_in\\">document</span>.location.href)
  <span class=\\"hljs-built_in\\">console</span>.log(<span class=\\"hljs-string\\">\\"hello\\"</span>)
  <span class=\\"hljs-comment\\">// You can also write comments!</span>
}</code></pre>
  </section>
</article>"
`;
